outlines/structure.md

```diff
@@ -0,0 +1,56 @@
+# Application structure and code style
+
+1. What's different between Meteor and other kinds of apps?
+    1. All JavaScript, enables code sharing
+    2. Structure is flexible
+    3. Currently everything is about magic file names
+    4. App scope vs. package scope vs. file scope variables
+2. Client vs. server vs. both
+3. Smooth path from prototype to production, three sizes of apps, S, M, L
+4. General stuff
+    1. Filenames - named the same as the export
+        1. Every file exports one thing
+    2. Avoid depending on load order via namespacing and Meteor.startup
+        1. If you depend on load order, your best bet is a package
+    3. Naming guidelines for everything
+        1. https://github.com/meteor/guide/issues/9#issuecomment-148800085
```

Comment:
Yea, I think we are introducing too many new concepts here. `new Method`? I like the idea that it would be like that, but it is not. We are making guide for what we have currently, not what we would want in the future to be. We can do that in iterations and improve in the future.

BTW, I did something similar in [this package](https://github.com/peerlibrary/meteor-middleware), made publish endpoints `new PublishEndpoint`. The idea is that this allows you (later on) to attach also documentation to the publish endpoint. And also stack transformations.

But currently Meteor is more functions and callbacks oriented, so let's describe that.